src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -28,19 +28,24 @@ internal static class Utf8Encoder
         private const byte b1111_0000U = 0xF0; //240
         private const byte b1111_1000U = 0xF8; //248
 
+        // bits 30 & 31 are used for pending bits fixup
```

Comment:
This is used to deal with partially encoded characters.

You should not need this because of the corefx decoders do not ever stop at half character. Instead, they stop will not fully consume the input buffer.